src/main/java/com/cloudant/fdblucene/FDBLock.java

```diff
@@ -17,19 +17,39 @@
 
 import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.UUID;
 
 import org.apache.lucene.store.AlreadyClosedException;
 import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
 import org.apache.lucene.store.Lock;
+import org.apache.lucene.store.LockObtainFailedException;
 
 final class FDBLock extends Lock {
 
     private final Directory dir;
+    private final UUID uuid;
     private final String name;
+    private boolean invalid = false;
     private boolean closed = false;
 
-    FDBLock(final Directory dir, final String name) {
+    public static Lock obtain(final Directory dir, final UUID uuid, final String name) throws IOException {
+        try {
+            try (final IndexOutput output = dir.createOutput(name, null)) {
```

Comment:
this nested try with out catch looks weird

Reply:
that is the java 8 AutoCloseable thing. It means output.close() will be called when scope ends.